examples/http/client.cc

```diff
@@ -3,9 +3,15 @@
 
 #include "opentelemetry/ext/http/client/http_client_factory.h"
 #include "opentelemetry/ext/http/common/url_parser.h"
-#include "opentelemetry/trace/experimental_semantic_conventions.h"
 #include "tracer_common.h"
 
+#ifdef DELETE
+// Causes a build error with FaasDocumentOperationValues::DELETE
+#  undef DELETE
+#endif
```

Comment:
Can we include this ifdef/undef in template file, so we won't have to include it other places?

Reply:
Hi @lalitb 

Thanks for the review.

I considered this, but I am reluctant to add workarounds in the generated code for all users and all platforms to see, while the issue root cause is only for one platform (windows) and one header file (still to be identified) that pollutes the build.

If we add such code, it will never be removed.

Instead, I hope we can identify where this define DELETE is coming from, and put the work around closer to the root cause, removing this from examples.

No code in opentelemetry-cpp defines DELETE, so this most likely comes from a third party or from the system headers.

Having XXXFactory::Create() methods (#1393) will, as a side effect, remove a lot of header dependencies, possibly removing the root cause as well, so this work around should be temporary.

Not having windows locally, I can't tell which header is causing this, so anyone jumping in to investigate will be most welcome.

Reply:
Keeping this item unresolved, so we do not forget about it.

@lalitb Please merge #1393 to the repo, so this PR and the need for the windows work around can be re-evaluated after 1393.

Reply:
FYI, `DELETE` is defined as macro with mask value in `winnt.h`.

```c++
#define DELETE                           (0x00010000L)
```

Reply:
@ThomsonTan Thanks for finding the offending header.

@lalitb Updated the PR today to:

- implement a work around only for windows (_WIN32)
- add a compilation message when the code uses undef DELETE, to avoid surprises

Given how many include chain can land in winnt.h, and given how this file can be included indirectly by generated code with gRPC, having the work around in 4 places in examples is the best we can do, and the most robust for now.

Being in examples, this work around also will have value for users, to show how to resolve the build break.

So, proposing to merge the PR as it stands now, with the following code:

```
#ifdef _WIN32
#  ifdef DELETE
// winnt.h defines DELETE
// Causes a build error with FaasDocumentOperationValues::DELETE
#    pragma message(__FILE__ ": removing define on DELETE")
#    undef DELETE
#  endif
#endif

```

Ok to merge.

Please squash all the commit messages when merging.